main/src/library/Tcp.flix

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright 2025 Ry Wiese
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import java.net.ServerSocket
+
+enum TcpServer(ServerSocket)
+
+///
+/// The effect used to bind a TCP server.
+///
+eff Tcp {
+    def bind(ip: IpAddr, port: Int16): TcpServer
+}
+
+mod Tcp {
+    use IoError.ErrorKind
+    use IoError.IoError
+
+    import java.io.IOException
+    import java.lang.IllegalArgumentException
+    import java.lang.SecurityException
+    import java.net.InetAddress
+    import java.net.ServerSocket
+
+    ///
+    /// Handles the `Tcp` effect of the given function `f`.
+    ///
+    /// In other words, re-interprets the `Tcp` effect using the `Net` and `IO` effects.
+    ///
+    pub def handle(f: a -> b \ ef): a -> Result[IoError, b] \ {ef, IO, Net} - Tcp = x ->
+        run {
+            Ok(f(x))
+        } with handler Tcp {
+            def bind(ip, port, resume) = {
+                try {
+                    let inet = IpAddr.toInetAddress(ip);
+                    let port32 = Int16.toInt32(port);
+                    let socket = new ServerSocket(port32, 50, inet);
+                    let server: TcpServer = TcpServer.TcpServer(socket);
+                    resume(server)
+                } catch {
+                    case ex: SecurityException        => Err(IoError(ErrorKind.Security, ex.getMessage()))
```

Comment:
This line can be dropped. SecurityManager is deprecated and will soon be gone. Besides no one uses it.